### PR TITLE
Rem 450

### PR DESCRIPTION
**Pull request by FatimaSalami**
in this task physical-count-item is migrated from asp
metal summary control also is migrated
this control opens from a button and requests metalView then depending on the current grid's data shows a metal summary with calculations.. (you can compare with asp)

### CODE REVIEW

File: src/components/Shared/MetalSummary.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        const viewedList = processMetalsList(updatedData)
+
+        const transformedData = {
+          count: viewedList.length,
+          list: viewedList
+        }
+
+        setGridData(transformedData)
+      })
+      .catch(error => {})
```
Comment (RebeccaFrem02) [inline src/components/Shared/MetalSummary.js:42]: remove catch

File: src/components/Shared/MetalSummary.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  const processMetalsList = metalsList => {
+    // Update weight based on priceType
```
Comment (RebeccaFrem02) [inline src/components/Shared/MetalSummary.js:46]: remove comments

File: src/components/Shared/MetalSummary.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      headerName: labels.weight,
+      flex: 1
+    }
+  ]
+
+  return (
+    <VertLayout>
+      <Grow>
+        <Table
+          sx={{ mb: 3 }}
```
Comment (RebeccaFrem02) [inline src/components/Shared/MetalSummary.js:119]: is there a need for the styling here?

File: src/pages/physical-count-item/index.js [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+      headerName: _labels.weight,
+      flex: 1
+    }
+  ]
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        await fetchGridData()
+      } catch (error) {}
```
Comment (RebeccaFrem02) [inline src/pages/physical-count-item/index.js:134]: remove try catch

File: src/pages/physical-count-item/index.js [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+import { ControlContext } from 'src/providers/ControlContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { useWindow } from 'src/windows'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SCRepository } from 'src/repositories/SCRepository'
+import FormShell from 'src/components/Shared/FormShell'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { useForm } from 'src/hooks/form'
+import * as yup from 'yup'
+import MetalSummary from 'src/components/Shared/MetalSummary'
```
Comment (RebeccaFrem02) [inline src/pages/physical-count-item/index.js:18]: no need for the import here

File: src/pages/physical-count-item/index.js [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    setSiteStore([])
+    var parameters = `_stockCountId=${stockCountId}`
+    getRequest({
+      extension: SCRepository.Sites.qry,
+      parameters: parameters
+    })
+      .then(res => {
+        setSiteStore(res.list)
+      })
+      .catch(error => {})
```
Comment (RebeccaFrem02) [inline src/pages/physical-count-item/index.js:89]: remove all try chatches

File: src/components/Shared/MetalSummary.js (hunk 2 above)
Comment (FatimaSalami) [reply src/components/Shared/MetalSummary.js:46]: i kept them on purpose to have clear logic

File: src/components/Shared/FormShell.js [hunk 7]
```diff
@@ -290,6 +294,18 @@ export default function FormShell({
               title: platformLabels.Aging
             })
           }
+          onClickMetal={() =>
+            stack({
+              Component: MetalSummary,
+              props: {
+                originalList: filteredItems
+              },
```
Comment (Alinadimmokdad) [inline src/components/Shared/FormShell.js:302]: if you want you can directly send the filtered item and use it in metal summary

File: src/resources/ResourceIds.js [hunk 8]
```diff
@@ -269,6 +269,8 @@ export const ResourceIds = {
   EmployeeFilter: 71201,
   FinancialIntegrators: 51105,
   SalesDefaults: 51114,
+  IVPhysicalCountItem: 44202,
+  FE_MetalSummaryControl: 41601, //Metal button,
   FIAging: 31405,
```
Comment (Alinadimmokdad) [inline src/resources/ResourceIds.js:274]: rem comment

**Review by Alinadimmokdad — APPROVED** [review]

**Pull request merged**